src/main.rs

```diff
@@ -494,31 +520,42 @@ async fn get_last_block_height(
 ///
 /// The return value of [`None`] means there are no understandable CIS2 logs
 /// produced.
-fn get_cis2_events(bi: &BlockItemSummary) -> Option<Vec<(ContractAddress, Vec<cis2::Event>)>> {
+fn get_cis2_events(bi: &BlockItemSummary) -> Result<Option<ContractEffects>, DatabaseError> {
     match bi.contract_update_logs() {
-        Some(log_iter) => Some(
-            log_iter
-                .flat_map(|(ca, logs)| {
-                    match logs
-                        .iter()
-                        .map(cis2::Event::try_from)
-                        .collect::<Result<Vec<cis2::Event>, _>>()
-                    {
-                        Ok(events) => Some((ca, events)),
-                        Err(_) => None,
+        Some(log_iter) => {
+            // Map each log into a Result
+            let events: Result<Vec<_>, _> = log_iter
+                .map(|upward| match upward {
+                    Upward::Known((ca, logs)) => {
+                        let evs: Result<Vec<_>, _> =
+                            logs.iter().map(cis2::Event::try_from).collect();
```

Comment:
If you have transaction logs here (that are not smart contract cis2 logs) e.g. a  contract not related to tokens, then you will try to parse the logs into cis2 events and fail the parsing and return an error which will stop the indexer from progressing. I don't think that is what we want.